main.go

```diff
@@ -13,129 +11,183 @@ import (
-					}
-					stmt := ast.Statements[0]
-					if ast.Errors[0] != nil {
-						return reject.Invoke(Error.New("non sysntax error encountered: " + ast.Errors[0].Error()))
-					}
-					if _, ok := stmt.(sqlparser.CreateTableStatement); !ok {
-						return reject.Invoke(Error.New("the query isn't a CREATE"))
-					}
-					node := stmt.(*sqlparser.CreateTable)
-					if !createTableNameRegEx.MatchString(node.Table.String()) {
-						return reject.Invoke(Error.New("the query references a table name with the wrong format"))
-					}
-					parts := strings.Split(node.Table.String(), "_")
-					if len(parts) < 2 {
-						return reject.Invoke(Error.New("table name isn't referencing the chain id"))
+			return resolve.Invoke(js.ValueOf(response))
+		}()
+		return nil
+	})
+	return Promise.New(handler)
+}
+
+func normalize(this js.Value, args []js.Value) interface{} {
+	Error := js.Global().Get("Error")
+	Promise := js.Global().Get("Promise")
+	if len(args) < 1 {
+		return Promise.Call("reject", Error.New("missing required argument: statement"))
+	}
+	statement := args[0].String()
+	var nameMap js.Value
+	if len(args) == 2 && args[1].Type() == js.TypeObject {
+		nameMap = args[1]
+	}
+	handler := js.FuncOf(func(this js.Value, args []js.Value) interface{} {
+		resolve := args[0]
+		reject := args[1]
+		go func() interface{} {
+			ast, err := sqlparser.Parse(statement)
+			if err != nil {
+				return reject.Invoke(Error.New("error parsing statement: " + err.Error()))
+			}
+			if len(ast.Statements) == 0 {
+				return reject.Invoke(Error.New("error parsing statement: empty string"))
+			}
+			if !nameMap.IsUndefined() {
+				if _, err := UpdateTableNames(ast, func(name string) (string, bool) {
+					value := nameMap.Get(name)
+					if value.IsUndefined() {
+						return "", false
 					}
-					_, err = strconv.ParseInt(parts[len(parts)-1], 10, 64)
-					if err != nil {
-						return reject.Invoke(Error.New("parsing chain id in table name: " + err.Error()))
+					return value.String(), true
+				}); err != nil {
+					return reject.Invoke(Error.New("error updating statement: " + err.Error()))
+				}
+			}
+			if len(ast.String()) > maxQuerySize {
+				return reject.Invoke(Error.New("statement size error: larger than specified max"))
+			}
+			statements := make([]interface{}, len(ast.Statements))
+			var statementType StatementType
+			for i, stmt := range ast.Statements {
+				switch stmt.(type) {
+				case sqlparser.CreateTableStatement:
+					statementType = Create
+				case sqlparser.ReadStatement:
+					statementType = Read
+				case sqlparser.GrantOrRevokeStatement:
+					if statementType == "" {
+						statementType = Acl
 					}
-					return resolve.Invoke(js.ValueOf(node.StructureHash()))
-				}()
-				return nil
-			})
-			return js.Global().Get("Promise").New(handler)
-		}),
+				case sqlparser.WriteStatement:
+					statementType = Write
+				}
+				statements[i] = stmt.String()
+			}
+			tableReferences := sqlparser.GetUniqueTableReferences(ast)
```

Comment:
Is the first table in this response _always_ the target/primary table? This is a question for @brunocalza:
Is the ordering in the list of tables based on anything, or just order encountered? I'm hoping there is some common pattern for creates (always the first) and insert/updates (always the first), whereas for reads it doesn't _really_ matter...

Reply:
It is based on the order it encounters. But most of the statements have only one table (`CREATE`, `GRANT`, `REVOKE`, `DELETE`, and `UPDATE`). `INSERT` can have at most two: one for the `INSERT`and one for a possible `SELECT`.  `SELECT` can have many, and it will be in the order the walker walks.